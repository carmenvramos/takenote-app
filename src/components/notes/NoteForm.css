.NotesForm {
  margin: 1em auto;
  width: 50%;
  max-width: 500px;

   input {
    display: block;
    font-size: 1.25em;
    align-self: left;
    width: 50%;
    max-width: 500px;
    margin: auto;
    padding: 0.6em;
    border: 3mm ridge rgba(52, 21, 114, 0.618);
  }

   textarea {
    display: block;
    font-size: 1.25em;
    width: 100%;
    max-width: 470px;
    height: 100vh;
    max-height: 500px;
    margin: 0.5em;
    padding: 0.4em;
    border: .5rem solid;
    border-color: #B16FB1;

  }

   button {
    display: block;
    width: 100%;
    margin: 0.6em;
    padding: 0.5em;
    border-radius: 10px;

    background-color: #2D1050;
    color: #E9E2E4;
    font-size: 1em;
    font-weight: 700;
  }
}
  
