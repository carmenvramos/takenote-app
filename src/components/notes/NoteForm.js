import React from 'react';
import PropTypes from 'prop-types';
import styles from './NoteForm.css';

function NotesForm({ title, body, createNote, titleChange, bodyChange }) {
  return (
    <form className={styles.NotesForm} onSubmit={createNote.bind(null, title, body)}>
      <input
        type="text"
        name="title"
        placeholder="Note Title"
        value={title}
        onChange={titleChange}
      />
      <textarea
        name="body" value={body}
        onChange={bodyChange}
      ></textarea>
      <button>Create</button>
    </form>
  );
}

NotesForm.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  createNote: PropTypes.func.isRequired,
  titleChange: PropTypes.func.isRequired,
  bodyChange: PropTypes.func.isRequired
};

export default NotesForm;
