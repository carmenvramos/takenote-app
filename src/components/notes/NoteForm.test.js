import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

describe('NoteForm tests', () => {
  let wrapper;
  let createNote;
  beforeEach(() => {
    createNote = jest.fn();
    wrapper = shallow(
      <NoteForm
        title="Note Title"
        body="Note Body"
        error={null}
        loading={false}
        createNote={createNote}
        titleChange={jest.fn()}
        bodyChange={jest.fn()}
      />
    );
  });
  it('matches a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the createNote function on submit', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });

    expect(createNote).toHaveBeenCalled();
  });
});
  
