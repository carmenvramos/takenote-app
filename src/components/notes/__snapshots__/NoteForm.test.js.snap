// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm tests matches a snapshot 1`] = `
<form
  className="NotesForm"
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[MockFunction]}
    placeholder="Note Title"
    type="text"
    value="Note Title"
  />
  <textarea
    name="body"
    onChange={[MockFunction]}
    value="Note Body"
  />
  <button
    disabled={false}
  >
    <NoteSpinner
      loading={false}
    >
      <span>
        Create
      </span>
    </NoteSpinner>
  </button>
</form>
`;
